### PR TITLE
Added version flag in sandbox

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added version flag in the sandbox start
- Remove error handling in version command and only added error logs 

```bash
➜ flytectl  sandbox start --version=v0.13.0
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
Error: version flag only support v0.14.0+ flyte version
exit status 1

➜ flytectl  sandbox start --version=v0.14.0
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
🐋 pulling docker image cr.flyte.org/flyteorg/flyte-sandbox:dind
{"status":"Pulling from flyteorg/flyte-sandbox","id":"dind"}
{"status":"Digest: sha256:5d488bbeb8814b69f9ecf091bb6fdcbbff4827b482c90140c0fb5c5326ee2c00"}
{"status":"Status: Image is up to date for cr.flyte.org/flyteorg/flyte-sandbox:dind"}
🧑‍🏭 booting Flyte-sandbox container
92021-07-08T15:58:49.643206200Z Starting Docker daemon...

➜ flytectl  sandbox start
🧑‍🏭 Bootstrapping a brand new flyte cluster... 🔨 🔧
🐋 pulling docker image cr.flyte.org/flyteorg/flyte-sandbox:dind
{"status":"Pulling from flyteorg/flyte-sandbox","id":"dind"}
{"status":"Digest: sha256:5d488bbeb8814b69f9ecf091bb6fdcbbff4827b482c90140c0fb5c5326ee2c00"}
{"status":"Status: Image is up to date for cr.flyte.org/flyteorg/flyte-sandbox:dind"}
🧑‍🏭 booting Flyte-sandbox container
92021-07-08T15:58:49.643206200Z Starting Docker daemon...

 ➜ flytectl  version (Offline command without internet and sandbox)
Installed flytectl version is the latest
{
  "App": "flytectl",
  "Build": "4d6befc",
  "Version": "v0.1.29-12-g4d6befc",
  "BuildTime": "2021-07-08 21:24:14.727836 +0530 IST m=+0.018597084"
}     
```
## Type
- [x] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [x] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by kumare3 — DISMISSED** [review]

File: pkg/util/util.go [hunk 1]
```diff
@@ -5,24 +5,28 @@ import (
...
-		return []byte(""), err
+	if response.StatusCode == 200 {
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return []byte(""), err
+		}
+		return data, nil
 	}
-	return data, nil
+	return []byte(""), fmt.Errorf("something goes wrong, Got status code %v while sending request to %s", response.StatusCode, fmt.Sprintf("%s%s", baseURL, url))
```
Comment (EngHabu) [inline pkg/util/util.go:29]:
```suggestion
	return []byte(""), fmt.Errorf("something went wrong. Received status code [%v] while sending a request to [%s]", response.StatusCode, fmt.Sprintf("%s%s", baseURL, url))
```

Comment (EngHabu) [reply pkg/util/util.go:29]: Let's also move the `fmt.Sprintf()` part to the first line of this function and just use it where needed...

File: pkg/util/util.go [hunk 2]
```diff
@@ -41,3 +45,15 @@ func WriteIntoFile(data []byte, file string) error {
 	}
 	return nil
 }
+
+func CompareVersion(version1, version2 string) (bool, error) {
```
Comment (EngHabu) [inline pkg/util/util.go:49]:
```suggestion
func IsVersionGreaterThan(version1, version2 string) (bool, error) {
```

File: cmd/sandbox/start.go [hunk 3]
```diff
@@ -111,3 +120,43 @@ func startSandbox(ctx context.Context, cli docker.Docker, reader io.Reader) (*bu
...
+func mountFlyteManifest(version string) error {
+	if len(version) > 0 {
+		isGreater, err := util.IsVersionGreaterThan(version, FlyteMinimumVersionSupported)
+		if err != nil {
+			return err
+		}
+		if !isGreater {
+			return fmt.Errorf("version flag only support %s+ flyte version", FlyteMinimumVersionSupported)
+		}
+		response, err := util.GetRequest("https://github.com", fmt.Sprintf(FlyteReleaseURL, version))
```
Comment (EngHabu) [inline cmd/sandbox/start.go:148]: Let's move into const too...

File: pkg/util/util_test.go [hunk 4]
```diff
@@ -49,3 +54,32 @@ func TestWriteIntoFile(t *testing.T) {
 		assert.NotNil(t, err)
 	})
 }
+
+func TestCompareVersion(t *testing.T) {
```
Comment (EngHabu) [inline pkg/util/util_test.go:58]:
```suggestion
func TestIsVersionGreaterThan(t *testing.T) {
```

**Review by EngHabu — COMMENTED** [review]
a couple of nits... LGTM otherwise

**Review by pmahindrakar-oss — APPROVED** [review]

**Pull request merged**